src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/PropertyInfoPassTest.php

```diff
@@ -11,6 +11,7 @@
 
 namespace Symfony\Bundle\FrameworkBundle\Tests\DependencyInjection\Compiler;
 
+use Prophecy\Argument;
```

Comment:
Yes, the only problem with PHPunit mocks here is that `hasDefinition` is expected 2 times with different arguments. PHPunit does not support multiple expectations for the same method call in a very nice way. This means using something like:
```php
$container->expects($this->eq(0))
    ->method('hasDefinition')
    ->with('property_info')
    ->will($this->returnValue(true));
$container->expects($this->eq(1))
    ->method('hasDefinition')
    ->with('serializer.mapping.class_metadata_factory')
    ->will($this->returnValue(true));
```